src/core/control/Control.cpp

```diff
@@ -1584,6 +1586,47 @@ bool Control::openPdfFile(fs::path filepath, bool attachToDocument, int scrollTo
     return success;
 }
 
+bool Control::openPngFile(fs::path filepath, bool attachToDocument, int scrollToPage) {
+    fs::path imagePath(filepath);
+    this->getCursor()->setCursorBusy(true);
+    auto doc = std::make_unique<Document>(this);
+    this->replaceDocument(createNewDocument(this, std::move(filepath), std::nullopt), -1);
+
+    // Put a png file directly in the page
+    this->doc->lock();
+    const size_t pageNr = this->getCurrentPageNo();
+    PageRef page = this->doc->getPage(pageNr);
+    BackgroundImage img;
+    GError* error = nullptr;
+    img.loadFile(imagePath, &error);
+
+    if (error) {
+        std::string msg = FS(_F("Error reading PNG file \"{1}\"\n{2}") % imagePath.u8string() % doc->getLastErrorMsg());
```

Comment:
`doc->getLastErrorMsg()` only contains something when there is an error from reading a PDF
```suggestion
        std::string msg = FS(_F("Error reading PNG file \"{1}\"") % imagePath.u8string());
```